### DISCUSSION TITLE
Would using the preload_modules option in forkserver reduce my bootstrapping time? If not, how can I reduce my bootstrapping?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18015] [Q&A] [answered]
i noticed, there's a preload_modules in forkserver option, but didn't see any docs for it - would that help reduce this bootstrapping time?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1699989961088019?thread_ts=1699989961.088019&cid=C01U954MEER).*

**Comment by alangenfeld** [comment, marked as answer]
There is some documentation in the config schema description https://github.com/dagster-io/dagster/blob/1.5.7/python_modules/dagster/dagster/_core/definitions/executor_definition.py#L373-L377

Under the hood its using https://docs.python.org/3/library/multiprocessing.html#multiprocessing.set_forkserver_preload

> would that help reduce this bootstrapping time?

Potentially yes, if the import of certain modules is expensive then only importing them once in the forkserver template process will avoid having to pay the import cost again in subsequent forked processes.
